### PR TITLE
Bug 13123 - empty command line on MovementMarkable trait no longer put…

### PR DESCRIPTION
**Pull request by Cattlesquat**
…s blank line in popup menu

### CODE REVIEW

File: src/VASSAL/counters/MovementMarkable.java [hunk 1]
```diff
@@ -115,7 +117,15 @@ public String myGetType() {
...
-    return new KeyCommand[]{new KeyCommand(command, key, Decorator.getOutermost(this), this)};
+    if (kc == null) {  //BR// Bug 13123 - perform proper operation to make sure blank line isn't added to popup menu if command line string is blank
+      kc = new KeyCommand(command, key, Decorator.getOutermost(this), this);
+      if (command.length() > 0 && key != null && ! key.isNull()) {
+        kclist = new KeyCommand[]{kc};
+      } else {
+        kclist = new KeyCommand[0];
+      }
+    }
+    return kclist;
```
Comment (uckelman) [inline src/VASSAL/counters/MovementMarkable.java:128]: It's potentially problematic to store these, as `KeyCommand` isn't immutable, so whatever you hand off the key command array to could modify the `KeyCommand` in it, thus modifying the very `KeyCommand` you've kept here to hand out next time.

This way avoids that:
```
return command.isEmpty() || key == null || key.isNull() ?
  new KeyCommand[0] :
  new KeyCommand[]{
    new KeyCommand(command, key, Decorator.getOutermost(this), this)
};
```

Comment (Cattlesquat) [reply src/VASSAL/counters/MovementMarkable.java]: Sounds good. I was copying the "form" I found in Clone.java but that *totally* doesn't mean it was a good idea :-)  This is clearly better - change added to PR.

**Comment by uckelman** [comment]
Removing the `.classpath` file is going to screw things up for @BrentEaston while we're waiting for him to have a look at PR #27, so instead of merging this directly I cherry-picked the relevant commits from it. (So, your changes for the Bug are in master now, but I'm closing the PR without merging it.)

**Comment by BrentEaston** [comment]
I am trying out Yan's maven build branch now.

**Comment by Cattlesquat** [comment]
Oh sorry I thought I was deleting .classpath out of the commit, not out of the project.

**Comment by uckelman** [comment]
That's what I figured. It's a bit more complicated to restore a file you've deleted.

**Pull request closed without merging**